server/clustering_test.go

```diff
@@ -618,106 +598,123 @@ func TestClusteringLogSnapshotCatchup(t *testing.T) {
 // Ensures subscriptions are replicated such that when a leader fails over, the
 // subscription continues to deliver messages.
 func TestClusteringSubscriberFailover(t *testing.T) {
-	cleanupDatastore(t)
-	defer cleanupDatastore(t)
-	cleanupRaftLog(t)
-	defer cleanupRaftLog(t)
-
-	// For this test, use a central NATS server.
-	ns := natsdTest.RunDefaultServer()
-	defer ns.Shutdown()
-
-	// Configure first server
-	s1sOpts := getTestDefaultOptsForClustering("a", []string{"b", "c"})
-	s1 := runServerWithOpts(t, s1sOpts, nil)
-	defer s1.Shutdown()
-
-	// Configure second server.
-	s2sOpts := getTestDefaultOptsForClustering("b", []string{"a", "c"})
-	s2 := runServerWithOpts(t, s2sOpts, nil)
-	defer s2.Shutdown()
-
-	// Configure third server.
-	s3sOpts := getTestDefaultOptsForClustering("c", []string{"a", "b"})
-	s3 := runServerWithOpts(t, s3sOpts, nil)
-	defer s3.Shutdown()
+	testCases := []struct {
```

Comment:
You could have added the queue subscription case here too. `testCases` could be an array of `func() error` or simply `func()` if you use `t` to fail inside the function. The function can create the [durable|queue] subscription without having to return the sub (since the Unsubscribe is not required), but if you really want you could return an array of subs.
You could use `var sc stan.Conn, etc..` at top of function so that you can use the variables and not have to pass them as a param.